### DISCUSSION TITLE
Transcode only if certain cases

### DISCUSSION THREAD

**Discussion by soyelmismo** [navidrome/navidrome#2598] [Ideas]
Example: I dont want to transcode anything, but my browser stucks the playback when it tries to play a dsd file... for that, i added a wav transcoding that will convert everything into wav for playback but its a heavy task... so it could be a good practice to add a "transcoding case" like if the file to play is dsd or dsf format/extension so it will transcode to the selected transcoding format :)

an A.I. gave me this but i cant add bash scripting in the transcoding command.

```
case "$ext" in
  dsd|dsf)
    ffmpeg -i "$input" -map 0:0 -v 0 -f wav -
    ;;
  *)
    echo "Archivo $input no es .dsd o .dsf, omitiendo"
    ;;
esac
```

extensions = "dsd|dsf|wavpak"
or `ffmpeg -i %s $(if [[ "%s" =~ \.($extensions)$ ]]; then echo "-map 0:0 -v 0 -f wav -"; fi) 2>/dev/null`

something like that, but just a verification in navidrome code, thing that i dont know how to do...

**Comment by metalheim** [comment]
I think in theory you could put your code in a shell script (.sh file) and put that in the transcoding command.

In general, it would be much cleaner if this is implemented properly, there are open issues already with #351 and #2051 that cover this
